internal/pkg/template/overrides.go

```diff
@@ -0,0 +1,100 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package template
+
+import (
+	"fmt"
+	"strings"
+	"unicode"
+)
+
+const (
+	cdkVersion              = "2.56.0"
+	cdkConstructsMinVersion = "10.0.0"
+	cdkTemplatesPath        = "overrides/cdk"
+)
+
+// CFNResource represents a resource rendered in a CloudFormation template.
+type CFNResource struct {
+	Type      CFNType
+	LogicalID string
+}
+
+type cfnResources []CFNResource
+
+// Types returns the list of unique CFN types.
+func (rs cfnResources) Types() []CFNType {
+	var uniqueTypes []CFNType
+	seen := make(map[CFNType]struct{})
+	for _, r := range rs {
+		if _, ok := seen[r.Type]; !ok {
+			uniqueTypes = append(uniqueTypes, r.Type)
+		}
+		seen[r.Type] = struct{}{}
+	}
+	return uniqueTypes
+}
+
+// CFNType is a CloudFormation resource type such as "AWS::ECS::Service".
+type CFNType string
```

Comment:
I feel like we need a mechanism to make sure the `string` _is_ a `CFNType`. Maybe regex validation etc, so that we won't cause panic when calling `ImportName()` etc.